### PR TITLE
Add a protobuf definition for CASObject, reorganize a bit

### PR DESCRIPTION
**Pull request by dmbryson**

### CODE REVIEW

File: Sources/LLBCASProtocol/Generated/CASProtocol/cas_object.pb.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+//// LLBPBCASObject represents the serialized from of CASObjects.  It encodes the
+//// combination of the raw data of the object and its dependendent references.
+public struct LLBPBCASObject {
+  // SwiftProtobuf.Message conformance is added in an extension below. See the
+  // `Message` and `Message+*Additions` files in the SwiftProtobuf library for
+  // methods supported on all messages.
+
+  //// The command line arguments to execute on the remote machine.
```
Comment (aciidb0mb3r) [inline Sources/LLBCASProtocol/Generated/CASProtocol/cas_object.pb.swift:37]: 🤔

Comment (dmbryson) [reply Sources/LLBCASProtocol/Generated/CASProtocol/cas_object.pb.swift:37]: well, thats a strange merge error...

File: Protos/module_map.asciipb [hunk 2]
```diff
@@ -1,3 +1,8 @@
+mapping {
+    module_name: "LLBCASProtocol"
+    proto_file_path: "CASProtocol/cas_object.proto"
+    proto_file_path: "CASProtocol/data_id.proto"
```
Comment (sergiocampama) [inline Protos/module_map.asciipb:4]: can you also delete the data_id line in the next section?

File: Sources/LLBUtil/CAS/CASProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import Foundation
+
+import llbuild2
+import LLBCASProtocol
+
+
+public extension LLBDataID {
+    var asProto: LLBPBDataID {
```
Comment (sergiocampama) [inline Sources/LLBUtil/CAS/CASProtocol.swift:16]: you can also do

```
return LLBPBDataID.with { $0.bytes = Data(self.bytes) }
```

for less code :)

File: Sources/LLBBuildSystemProtocol/DataID.swift [hunk 4]
```diff
@@ -6,9 +6,12 @@
 // See http://swift.org/LICENSE.txt for license information
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
-import llbuild2
 import Foundation
 
+import llbuild2
```
Comment (sergiocampama) [inline Sources/LLBBuildSystemProtocol/DataID.swift:11]: should this move to CASProtocol instead? it looks fairly related to the other method you  have

Comment (dmbryson) [reply Sources/LLBBuildSystemProtocol/DataID.swift:11]: Yeah, this whole thing got a little muddled.  I had written it before artifacts but hadn't pushed it.  That'll teach me to leave it unfinished overnight. 😉

**Pull request merged**